bestbook/templates/browse.html

```diff
@@ -1,47 +1,123 @@
-<h2>Browse recommendations</h2>
 
-{% set page = request.args.get("page", 0) | int %}
-{% set recs = models["recommendations"].paginate(page, is_approved=True) %}
+{% set votes = models["votes"].get_batch(models["votes"], recs, username=session.get("username")) %}
+
+{% if request.args.get('debug') %}
+  {{ votes }}
+{% endif %}
 
 {% for rec in recs %}
 <div class="block-rec">
   <div class="recommendation">
-    <div>
-      <a href="https://openlibrary.org/works/{{rec.winner.work_olid}}">
-        {% if recs.works|length > 0 and recs.works[rec.winner.work_olid].get('cover_url') %}
-        <img class="bookcover" src="{{recs.works[rec.winner.work_olid].get('cover_url')}}">
-        {% endif %}
-      </a>
+    {% set winner = recs.works[rec.winner.work_olid] if rec.winner.work_olid in recs.works else {} %}
+    <div class="recommendation--upvote">
+      {% set my_vote = votes.get('user', {}).get(rec.id) %}
+      <form method="POST" action="/api/recommendations/{{rec.id}}/votes">
+        <input type="hidden" name="value" value="true">
+        <input type="submit" value="&#9652;" class="nostyle {{ 'selected' if my_vote == 1 else '' }}"/>
+      </form>
+      <p class="upvotes">{{ votes['totals'].get(rec.id, {}).get('score', 0) }}</p>
+      <form method="POST" action="/api/recommendations/{{rec.id}}/votes">
+        <input type="hidden" name="value" value="false">
+        <input type="submit" value="&#9662;" class="nostyle {{ 'selected' if my_vote == -1 else '' }}"/>
+      </form>
```

Comment:
We may want to POST via AJAX to avoid navigating away from the page.

Reply:
definitely the case, will followup w/ another PR